File: server/transactions/types/request/create.go
package request

import (
	"financo/core/domain/types/nullable"
	"time"
)

type Create struct {
	IssuedAt     time.Time                `json:"issuedAt"`
	ExecutedAt   nullable.Type[time.Time] `json:"executedAt"`
	Notes        nullable.Type[string]    `json:"notes"`
	SourceID     int64                    `json:"sourceID"`
	TargetID     int64                    `json:"targetID"`
	SourceAmount int64                    `json:"sourceAmount"`
	TargetAmount int64                    `json:"targetAmount"`
}

File: webapp/app/modules/ledger/types/create.ts
import { Currency } from "~/modules/shared/types/currency"
import { Kind, Transaction } from "./transactions"

export type Create = {
  issuedAt: string // it is a date formatted YYYY-MM-DD
  executedAt?: string | null // it is a date formatted YYYY-MM-DD
  notes?: string | null
  currency: Currency
  sourceId: number
  targetId: number
  sourceAmount: number
  targetAmount: number
  kind: Kind
}

type OnCreateTransactionSuccess = () => void
type OnCreateTransactionFailure = () => void

export type CreateTransactionAction =
  (values: Create, success: OnCreateTransactionSuccess, failure: OnCreateTransactionFailure) => Promise<Transaction>
export type OnSubmitCreateTransactionAction =
  (values: Create) => Promise<void>
